dhis-2/dhis-api/src/main/java/org/hisp/dhis/indicator/IndicatorService.java

```diff
@@ -57,6 +57,24 @@ public interface IndicatorService {
 
   List<Indicator> getAssociatedIndicators(List<IndicatorType> indicatorTypes);
 
+  List<Indicator> getIndicatorsByUid(List<String> uids);
+
+  /**
+   * Get all {@link Indicator}s that have the uid contained in its numerator
+   *
+   * @param uid to search
+   * @return matching {@link Indicator}s
+   */
+  List<Indicator> getIndicatorsWithNumeratorContaining(String uid);
```

Comment:
FYI: We have a `UID` type that we try to slowly introduce. The service API methods would be a good place to start using it even if we do not use it in our model objects yet.

Reply:
i'll update this